astropy/modeling/core.py

```diff
@@ -2022,7 +2023,7 @@ def _initialize_parameters(self, args, kwargs):
                 model_set_axis = False
         else:
             if not (model_set_axis is False or
-                    (isinstance(model_set_axis, int) and
+                    (np.issubdtype(type(model_set_axis), np.integer) and
```

Comment:
What about just 
```suggestion
                    (isinstance(model_set_axis, numbers.Integral) and
```
with an `import numbers` at the top of the module?

Reply:
I would also suggest removing check for `bool`: I do not like (it would be "unnatural") to label axes using `bool` but, in Python world nothing is too crazy. That is, if it works it works.

Reply:
Though (wrt booleans):
```python
In [101]: isinstance(np.bool_(1), numbers.Number)
Out[101]: False

In [102]: isinstance(bool(1), numbers.Number)
Out[102]: True
```
which is kind of annoying

Reply:
Note that
```
In [11]:  np.issubdtype(type(np.bool_(1)), np.integer)
Out[11]:  False

In [12]:  np.issubdtype(type(bool(1)), np.integer)
Out[12]:  False
```

Which nicely covers the annoying case you described above. Moreover, it also makes the `bool` check unnecessary.

In light of this, would you suggest I propose that we keep my current check and remove the `bool` check. Do you agree?

Reply:
I like what you suggest

Reply:
Just pushed a commit addressing this conversation.